build/android.mk

```diff
@@ -256,17 +277,35 @@ $(call SRC_TO_OBJ,$(SRC)/Android/NativeI2CbaroListener.cpp): $(NATIVE_HEADERS)
 $(call SRC_TO_OBJ,$(SRC)/Android/NativeNunchuckListener.cpp): $(NATIVE_HEADERS)
 $(call SRC_TO_OBJ,$(SRC)/Android/NativeVoltageListener.cpp): $(NATIVE_HEADERS)
 
-ANDROID_LIB_BUILD = $(patsubst %,$(ANDROID_ABI_DIR)/lib%.so,$(ANDROID_LIB_NAMES))
+ifeq ($(ANDROID_LEGACY),y)
+  ANDROID_LIB_BUILD = $(patsubst %,$(ANDROID_ABI_DIR)/lib%.so,$(ANDROID_LIB_NAMES)_legacy)
+else
+  ANDROID_LIB_BUILD = $(patsubst %,$(ANDROID_ABI_DIR)/lib%.so,$(ANDROID_LIB_NAMES))
+endif
 $(ANDROID_LIB_BUILD): $(ANDROID_ABI_DIR)/lib%.so: $(ABI_BIN_DIR)/lib%.so | $(ANDROID_ABI_DIR)/dirstamp
 	$(Q)cp $< $@
 
+ifeq ($(ANDROID_INCLUDE_LEGACY_LIB),y)
+  ANDROID_LEGACY_LIB_DIR = $(TARGET_OUTPUT_DIR)/$(XCSOAR_ABI)_legacy/bin
```

Comment:
I don't like that the formula for this variable has been implemented already elsewhere; this kind of code duplicate is fragile.